src/main/java/vct/col/util/AbstractTypeCheck.java

```diff
@@ -916,6 +918,15 @@ public void visit(OperatorExpression e) {
       case Unfolding: {
         if (!tt[0].isResource()) Fail("Cannot unfold type %s", tt[0]);
         e.setType(tt[1]);
+        MethodInvokation innerMi = getMethodInvokationInsideScale(operatorArgs[0]);
+        if (!(innerMi instanceof MethodInvokation)) {
```

Comment:
innerMi is of type MethodInvokation. Why would you check if it is an instanceof MethodInvokation? The only moment !(innerMi instanceof MethodInvokation) is true is when innerMi is null.